### ISSUE TITLE
feat: recognize OWNERS.yaml and OWNERS as known extraneous files

### ISSUE THREAD

**Issue by mpolce-squareup** [agent-ecosystem/skill-validator#33]
## Summary

`OWNERS.yaml` and `OWNERS` are repo ownership/governance files used across GitHub repositories (similar in purpose to `CODEOWNERS`) to assign reviewers and teams to directories. They have no relevance to agent skill execution.

Currently, skills containing these files at the root receive the generic fallback warning:

> `unexpected file at root: OWNERS.yaml — if agents need this file, move it into references/ or assets/...`

Adding them to `knownExtraneousFiles` in `structure/checks.go` would produce the more accurate message:

> `OWNERS.yaml is not needed in a skill — agents may load it into their context window, taking space from your actual task`

## Evidence

Tested against a skill registry with 375+ skills — `OWNERS.yaml` appears in **242 of 375 skills (65%)**, making it the most common source of the "unexpected file at root" warning by a wide margin. It is never referenced from `SKILL.md` as agent-relevant content.

For comparison, similar repo-management files already recognized in `knownExtraneousFiles`:
`README.md`, `CHANGELOG.md`, `LICENSE`, `CONTRIBUTING.md`, `Makefile`, `.gitignore`

## Proposed change

```go
// structure/checks.go
var knownExtraneousFiles = map[string]string{
    // existing entries ...
    "owners.yaml": "OWNERS.yaml",
    "owners":      "OWNERS",
}
```

No new flag or CLI changes needed.
